### PR TITLE
Merged DBV2 code from Master19 into daffy

### PR DESCRIPTION
**Pull request by AmeliaEScott**
This code was written by Merlin Hosner (@hosnerm), with minor changes by Timothy Scott (@ItsTimmy) when porting to daffy.

### CODE REVIEW

File: packages/duckietown_demos_dbv2/.autogenerated.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+Maintainer: [Liam Paull](mailto:lpaull@mit.edu)
+
+### Description {nonumber="1"}
+
+TODO: description of `duckietown_demos`
+
+
+
+</div>
+
```
Comment (AmaurX) [inline packages/duckietown_demos_dbv2/.autogenerated.md]: Is that file needed ? This seems to be and old autoggenerated file that we dont use anymore. I would suggest removing it, as it contains outdated info

File: packages/duckietown_demos_dbv2/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Package `duckietown_demos` {#duckietown_demos}
+
+<move-here src='#duckietown_demos-autogenerated'/>
```
Comment (AmaurX) [inline packages/duckietown_demos_dbv2/README.md]: Here, related to my other comment, I would just have a regular README file

File: packages/duckietown_demos_dbv2/launch/lane_following.launch [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<launch>
+	<arg name="veh" default="$(env VEHICLE_NAME)"/>
+	<arg name="ai_trafo_mode" default="cb" doc="'cb' for colo balance only; 'both' for color balance and linear trafo"/>
+	<arg name="ai_interval" default="5" doc="interval with which the linear trafo gets updated. color balance is performed every second."/>
```
Comment (AmaurX) [inline packages/duckietown_demos_dbv2/launch/lane_following.launch:4]: So those are here, are they not needed for the indefinite navigation demo as well then ?

Comment (AmaurX) [reply packages/duckietown_demos_dbv2/launch/lane_following.launch:4]: I mean ai_interval, ai_trafo_mode and such

File: packages/duckietown_demos_dbv2/package.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<package>
+  <name>duckietown_demos_dbv2</name>
+  <version>1.0.0</version>
+  <description>TODO: description of `duckietown_demos`</description>
+
+  <maintainer email="lpaull@mit.edu">Liam Paull</maintainer>
```
Comment (AmaurX) [inline packages/duckietown_demos_dbv2/package.xml]: please update description and maintainer

File: packages/lane_control_dbv2/.autogenerated.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+TODO: Missing description for entry "`fsm_mode`".
+
+### Publishers {nonumber="1"}
+
+**Publisher `car_cmd`**: topic `~car_cmd` (`Twist2DStamped`)
+
+TODO: Missing description for entry "`car_cmd`".
+
+
+
```
Comment (AmaurX) [inline packages/lane_control_dbv2/.autogenerated.md]: same comment as for above, remove it and actually have a readme

File: packages/lane_control_dbv2/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,18 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(lane_control_dbv2)
+
+
+find_package(catkin REQUIRED COMPONENTS
+  duckietown_msgs
+  geometry_msgs
+  roscpp
```
Comment (AmaurX) [inline packages/lane_control_dbv2/CMakeLists.txt]: roscpp is needed ?

File: packages/lane_control_dbv2/README.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+<move-here src='#lane_control-lane_controller_node-autogenerated'/>
+
+This node takes pose estimations and reference values, such as velocity and curvature, from different nodes. It decides on which values to use from which node, based on the current mode of the state machine and outputs the yaw rate omega. It thereby provides a controller for lane following and path following (e.g. on intersections and parking lots). The code for the input of some nodes is not yet activated (still commented out).
+
+* TODO: After doing the new kinematic calibration provided by the System Identification group:
+    - The controller parameters should be adjusted according to the output of the calibration.
+    - The correction factors **velocity_to_m_per_s** and **omega_to_rad_per_s** need to be adjusted or ideally become obsolete and thus need to be deleted.
+
+To run the code the lane following demo can be used.
+
```
Comment (AmaurX) [inline packages/lane_control_dbv2/README.md]: all of that seems outdated a lot.

File: packages/lane_control_dbv2/README.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    $ make demo-lane-following
```
Comment (AmaurX) [inline packages/lane_control_dbv2/README.md]: especially this !!

File: packages/lane_control_dbv2/config/lane_controller_node/default.yaml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+k_d: -0.3
+# k_theta_old: -6.0
+k_theta: -0.2
+k_Id: 0.3
+# =======
+# v_bar: 0.2
+# k_d: -7
+# k_theta: -2
+# k_Id: 0.5
+# >>>>>>> origin/devel-intersection_navigation-final-report
```
Comment (AmaurX) [inline packages/lane_control_dbv2/config/lane_controller_node/default.yaml]: please remove the old merge conflict artefacts

File: packages/lane_control_dbv2/package.xml [hunk 10]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<package>
+  <name>lane_control_dbv2</name>
+  <version>1.0.0</version>
+  <description>TODO: Add a description of package `lane_control` in `package.xml`.</description>
+
+  <author email="steven@todo.todo">steven</author>
+  <author email="syliu@mit.edu">Shih-Yuan Liu</author>
+  <maintainer email='paull@mit.edu'>Liam Paull</maintainer>
+
```
Comment (AmaurX) [inline packages/lane_control_dbv2/package.xml:10]: please update all that as well

File: packages/lane_control_dbv2/package.xml [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  <build_depend>geometry_msgs</build_depend>
+  <build_depend>roscpp</build_depend>
+  <build_depend>rospy</build_depend>
+  <build_depend>std_msgs</build_depend>
+  <run_depend>duckietown_msgs</run_depend>
+  <run_depend>geometry_msgs</run_depend>
+  <run_depend>roscpp</run_depend>
+  <run_depend>rospy</run_depend>
+  <run_depend>std_msgs</run_depend>
+
```
Comment (AmaurX) [inline packages/lane_control_dbv2/package.xml:19]: same comment about the "depend" to replace both "run depend" and "build depend"

**Review by AmaurX — CHANGES_REQUESTED** [review]
There are a lot of old things that should never arrive here, please update and remove

File: packages/duckietown_demos_dbv2/launch/lane_following.launch (hunk 3 above)
Comment (AmeliaEScott) [reply packages/duckietown_demos_dbv2/launch/lane_following.launch:4]: I don't know, I copied this launch file over from DBV1.
